packages/playground/src/components/Editor.tsx

```diff
@@ -8,50 +8,49 @@ const monacoEditorOptions = {
   automaticLayout: true,
 };
 
-const Editor: React.FC<{ title: string; code: string; onChange: (code: string) => void }> = memo(
-  ({ title, code: initialCode, onChange }) => {
-    const [valid, setValid] = useState(true);
-    const [code, setCode] = useState(initialCode);
+interface EditorProps {
+  title: string;
+  code: string;
+  onChange: (code: string) => void;
+}
 
-    const onCodeChange = useCallback(
-      (code: string | undefined) => {
-        if (!code) {
-          return;
-        }
+export default function Editor({ title, code, onChange }: EditorProps) {
+  const [valid, setValid] = useState(true);
 
-        try {
-          const parsedCode = JSON.parse(code);
-          setValid(true);
-          setCode(code);
-          onChange(parsedCode);
-        } catch (err) {
-          setValid(false);
-          setCode(code);
-        }
-      },
-      [setValid, setCode, onChange]
-    );
+  const onCodeChange = useCallback(
+    (code: string | undefined) => {
+      if (!code) {
+        return;
+      }
 
-    const icon = valid ? 'ok' : 'remove';
-    const cls = valid ? 'valid' : 'invalid';
+      try {
+        const parsedCode = JSON.parse(code);
+        setValid(true);
+        onChange(parsedCode);
+      } catch (err) {
+        setValid(false);
+      }
```

Comment:
I would think you would want to set state on any change (as before), or else code won't update while the JSON is invalid? 

Or does the editor have a hidden/local state value that we don't need to set?

```suggestion
      try {
        const parsedCode = JSON.parse(code);
        setValid(true);
      } catch (err) {
        setValid(false);
      } finally {
        onChange(parsedCode);
      }
```